### PR TITLE
Nexus docs

### PR DESCRIPTION
**Pull request by bergundy**
[Rendered](https://github.com/temporalio/temporal/blob/b0538909f8cd958c010bd565d0edb1bf82902f08/docs/architecture/nexus.md)

### CODE REVIEW

File: docs/architecture/nexus.md [hunk 1]
```diff
@@ -0,0 +1,448 @@
+# Background
+
+Nexus RPC is a modern open-source service framework for arbitrary-length operations whose lifetime may extend beyond a
```
[the remark this answers is not in the document]
Comment (bergundy) [reply docs/architecture/nexus.md]: I know your sentiment on this terminology. This is the text we've consistently been using in our RFCs and proposals. I'd rather not deviate now.

File: docs/architecture/nexus.md [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+traditional RPC. Nexus was designed with durable execution in mind, as an underpinning to connect durable executions
+within and across namespaces, clusters and regions – with a clean API contract to streamline multi-team collaboration.
+Any service can be exposed as a set of sync or async Nexus operations – the latter provides an operation identity and a
+uniform interface to get the status of an operation or its result, receive a completion callback, or cancel the
+operation.
+
+Temporal is leveraging the Nexus RPC protocol to facilitate calling across namespace and cluster and boundaries.
+
```
[the remark this answers is not in the document]
Comment (bergundy) [reply docs/architecture/nexus.md]: This works for me.

File: docs/architecture/nexus.md [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+2. Enable Nexus via dynamic config, set the public callback URL, and set the allowed callback addresses.
+
+    ```yaml
+    system.enableNexus:
+      - value: true
+    component.nexusoperations.callback.endpoint.template:
+      # The URL must be publicly accessible if the callback is meant to be called by external services.
+      # When using Nexus for cross namespace calls, the URL's host is irrelevant as the address is resolved using
+      # membership.
+      - value: https://$PUBLIC_URL:7243/namespaces/{{.NamespaceName}}/nexus/callback
```
[the remark this answers is not in the document]
Comment (bergundy) [reply docs/architecture/nexus.md:74]: This is literally a Go template embedded in the config value, it's a cluster global config. I'll clarify.

File: docs/architecture/nexus.md [hunk 4]
```diff
@@ -0,0 +1,448 @@
...
+shard, this is dynamically configurable via: `history.outboundQueueMaxReaderCount`.
+
+#### Disabling the Outbound Queue Processor
+
+The outbound queue reader can be disabled dynamically by setting `history.outboundTaskBatchSize` to `0` (default is
+`100`).
+
+### Scheduler
+
+The outound queue processor uses the [`GroupByScheduler`](https://github.com/temporalio/temporal/blob/a8799ae43286f7dddf3147439bc2129f25065456/common/tasks/group_by_scheduler.go#L49) to group tasks into a per source namespace and destination [`DynamicWorkerPoolScheduler`](https://github.com/temporalio/temporal/blob/a8799ae43286f7dddf3147439bc2129f25065456/common/tasks/dynamic_worker_pool_scheduler.go#L47).
```
[the remark this answers is not in the document]
Comment (bergundy) [reply docs/architecture/nexus.md:155]: The tasks are grouped into separate schedulers though...

File: docs/architecture/nexus.md [hunk 5]
```diff
@@ -0,0 +1,448 @@
...
+
+Each task within its group, goes through an in memory buffer, a concurrency limiter, a rate limiter, and a circuit breaker.
+
+The entire group's processing stack emits metrics with the group's key (the source namespace name and destination).
```
[the remark this answers is not in the document]
Comment (bergundy) [reply docs/architecture/nexus.md:159]: I mention name to highlight that we use the name and not the internal ID.

File: docs/architecture/nexus.md [hunk 6]
```diff
@@ -0,0 +1,448 @@
...
+        rpc:
+          # NOTE: keep other fields as they were
+          httpPort: 7243
+
+    clusterMetadata:
+      # NOTE: keep other fields as they were
+      clusterInformation:
+        active:
+          # NOTE: keep other fields as they were
+          httpAddress: $ADDRESS_ACCESSIBLE_TO_PEERS:7243
```
[the remark this answers is not in the document]
Comment (bergundy) [reply docs/architecture/nexus.md]: There's a distinction here. This address only needs to be available to peer clusters.

File: docs/architecture/nexus.md [hunk 7]
```diff
@@ -0,0 +1,447 @@
...
+
+2. Enable Nexus via dynamic config, set the public callback URL, and set the allowed callback addresses.
+
+    ```yaml
+    system.enableNexus:
+      - value: true
+    component.nexusoperations.callback.endpoint.template:
+      # The URL must be publicly accessible if the callback is meant to be called by external services.
+      # When using Nexus for cross namespace calls, the URL's host is irrelevant as the address is resolved using
+      # membership.
```
Comment (bergundy) [inline docs/architecture/nexus.md:73]:
```suggestion
      # membership. The URL is a Go template that interpolates the `NamepaceName` and `NamespaceID` variables.
```

**Review by dandavison — APPROVED** [review]
LGTM!

**Pull request merged**